zipline/finance/blotter.py

```diff
@@ -162,7 +162,8 @@ def execute_cancel_policy(self, event):
             warn = self.cancel_policy.warn_on_cancel
 
             for orders_by_sid in itervalues(self.open_orders):
-                for order in orders_by_sid:
+                # So we don't mutate the list as we cancel orders
+                for order in orders_by_sid[:]:
```

Comment:
https://github.com/quantopian/zipline/pull/1047/files#diff-9dc7cb37547c5feca2c0771447ce1a63R147 needs to be updated then.